gallery/gallery/lib/demos/material/sliders_demo.dart

```diff
@@ -0,0 +1,504 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+enum SlidersDemoType {
+  sliders,
+  rangeSliders,
+  customSliders,
+}
+
+class SlidersDemo extends StatelessWidget {
+  const SlidersDemo({Key key, this.type}) : super(key: key);
+
+  final SlidersDemoType type;
+
+  String _title(BuildContext context) {
+    switch (type) {
+      case SlidersDemoType.sliders:
+        return GalleryLocalizations.of(context).demoSlidersTitle;
+      case SlidersDemoType.rangeSliders:
+        return GalleryLocalizations.of(context).demoRangeSlidersTitle;
+      case SlidersDemoType.customSliders:
+        return GalleryLocalizations.of(context).demoCustomSlidersTitle;
+    }
+    return '';
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget sliders;
+    switch (type) {
+      case SlidersDemoType.sliders:
+        sliders = _Sliders();
+        break;
+      case SlidersDemoType.rangeSliders:
+        sliders = _RangeSliders();
+        break;
+      case SlidersDemoType.customSliders:
+        sliders = _CustomSliders();
+    }
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        title: Text(_title(context)),
+      ),
+      body: Theme(
+        // Set platform to display material sliders only.
+        data: Theme.of(context).copyWith(platform: TargetPlatform.android),
+        child: sliders,
+      ),
+    );
+  }
+}
+// BEGIN slidersDemo
+
+class _Sliders extends StatefulWidget {
+  @override
+  _SlidersState createState() => _SlidersState();
+}
+
+class _SlidersState extends State<_Sliders> {
+  double _continuousValue = 25.0;
+  double _discreteValue = 20.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 40.0),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              Semantics(
+                label: GalleryLocalizations.of(context)
+                    .demoSlidersEditableNumericalValue,
+                child: SizedBox(
+                  width: 64,
+                  height: 48,
+                  child: TextField(
+                    textAlign: TextAlign.center,
+                    onSubmitted: (value) {
+                      final double newValue = double.tryParse(value);
+                      if (newValue != null && newValue != _continuousValue) {
+                        setState(() {
+                          _continuousValue =
+                              newValue.clamp(0.0, 100.0) as double;
```

Comment:
If newValue is of type double, why do you need to cast here?

Reply:
Seems to be a dart issue:
https://github.com/dart-lang/sdk/issues/39652